site/docs/guide.md

```diff
@@ -327,20 +327,77 @@ be determined, for example to manually clean up the cache. The cache is never
-if no hash is specified). While the condition on the file name is not
-necessary for correctness, it reduces the number of candidate files to one per
-specified directory. In this way, specifying distribution files directories
-remains efficient, even if the number of files in such a directory grows
-large.
+Besides the repository cache, Bazel has a second mechanism to avoid unnecessary
+downloads, called the distribution directory.
+
+Using the
+[`--distdir=/path/to-directory`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--distdir)
+option, you can specify additional read-only directories to look for files
+instead of fetching them. A file is taken from such adirectory if the file name
+is equal to the base name of the URL and additionally the hash of the file is
+equal to the one specified in the download request (again, no file is taken from
+a distdir, if no hash is specified).
+
+While the condition on the file name is not necessary for correctness, it
+reduces the number of candidate files to one per specified directory. In this
+way, specifying distribution files directories remains efficient, even if the
+number of files in such a directory grows large.
+
+#### Running Bazel in an airgapped environment
+
+To keep Bazel's binary size small, Bazel's implicit dependencies are fetched
+over the network while running for the first time. These implicit dependencies
+contain toolchains and rules that may not be necessary for everyone. For
+example, Android tools are unbundled and fetched only when building Android
+projects.
+
+However, these implicit dependencies for Bazel may cause problems when running
+Bazel in an airgapped environment, even if you have vendored all of your
+WORKSPACE dependencies. To solve that, we can prepare a distribution directory
+containing these dependencies on a machine with network access, and then
+transfer them to the airgapped environment with an offline approach.
+
+To prepare the [distribution directory](distribution-files-directories), use the
+[`--distdir`](https://docs.bazel.build/versions/master/command-line-reference.html#flag--distdir)
+flag. You will need to do this once for every new Bazel binary version, since
+the implicit dependencies can be different for every release.
+
+To build these dependencies outside of your airgapped environment, first
+checkout the Bazel source tree at the right version:
+
+```
+git clone https://github.com/bazelbuild/bazel "$BAZEL_DIR" 
+cd "$BAZEL_DIR"
+git checkout "$BAZEL_VERSION" 
+```
+
+Then, build the tarball containing the implicit runtime dependencies for that
+specific Bazel version:
+
+```
+bazel build @additional_distfiles//:archives.tar
+```
+
+Export this tarball to a directory that can be copied into your airgapped
+environment. Note the `--strip-components` flag, because `--distdir` can be
+quite finicky with the directory nesting level:
+
+
+```
+tar xvf bazel-bin/external/additional_distfiles/archives.tar -C
```

Comment:
Perhaps split on two lines with a \\ at the end so it won't horizontal scroll and is still pastable.

Also, if you move --strip-compontns before -C NEWDIRECTORY, then someone can copy-paste the entire preamble and just type the distdir path.

Reply:
Resolving in the internal CL: cr/313751874